### PR TITLE
Split large topology into smaller 'vagrant up' batches (fixes #732)

### LINKED ISSUE #732: Launching VMs in batches
When trying to spin up large topologies (50+ VMs), the process is slow and random VMs often fail to boot with below error messages.

```
An error occurred while executing the action on the 'pe1'
machine. Please handle this error then try again:

Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.
```

```
If the box appears to be booting properly, you may want to increase
the timeout ("config.vm.boot_timeout") value.

An unexpected error occurred when executing the action on the
'pe12' machine. Please report this as a bug:

The specified wait_for timeout (2 seconds) was exceeded
```
It would be great if netlab could start VMs in batches - maybe it would ease the burden put on the CPU.

### PR DESCRIPTION
**Pull request by ipspace**
Documentation @ docs/labs/libvirt.md

@petercrocker This one is based on the discussions we had about Vagrant DHCP failures. Could you please test it out on one of your large topologies (please note that this is a parallel branch to the `multilab` one, so you can't use both features in parallel while testing it ;)

### CODE REVIEW

**Comment by petercrocker** [comment]
I tested this out. Settings:
```
defaults.providers.libvirt.batch_size: 10
defaults.providers.libvirt.batch_interval: 10
```

This created 3 batches of 10 devices per batch. 

```
~/labs/arista-ios-srx-sites ❯ vagrant status
Current machine states:

c1xr01                    running (libvirt)
c1xr02                    running (libvirt)
c1xr03                    running (libvirt)
c1xr04                    running (libvirt)
c1xr05                    running (libvirt)
c1xr06                    running (libvirt)
d1xr01                    running (libvirt)
d1xr02                    running (libvirt)
d2xr01                    running (libvirt)
d2xr02                    running (libvirt)
d3xr01                    running (libvirt)
d3xr02                    running (libvirt)
d1xfw01                   running (libvirt)
d2xfw01                   running (libvirt)
h1xgw1                    running (libvirt)
h1xgw2                    running (libvirt)
h1xsw01                   running (libvirt)
h1xsw02                   running (libvirt)
h1xsw03                   running (libvirt)
h1xsw04                   running (libvirt)
h2xgw1                    running (libvirt)
h2xgw2                    running (libvirt)
h2xsw01                   running (libvirt)
h2xsw02                   running (libvirt)
h2xsw03                   running (libvirt)
h2xsw04                   running (libvirt)
v1xsw01                   running (libvirt)
v1xsw02                   running (libvirt)
v1xsw03                   running (libvirt)
v1xsw04                   running (libvirt)
v1xsw05                   running (libvirt)
hostvsrx1                 running (libvirt)
hostvsrx2                 running (libvirt)
```


Note that a very minor issue with this approach is that some of the vagrant status messages end up being suppressed. It seems the next batch kicks in what the all of the batched VMs are booted, and "Waiting for SSH..." but then you don't see the vagrant messages that would normally occur after the VM gets a DHCP IP. 

For example:
```
==> d1xr01: Waiting for domain to get an IP address...
==> c1xr01: Waiting for SSH to become available...
==> c1xr04: Waiting for SSH to become available...
==> d2xr01: Waiting for SSH to become available...
==> c1xr05: Waiting for SSH to become available...
==> c1xr02: Waiting for SSH to become available...
==> d1xr02: Waiting for SSH to become available...
==> c1xr03: Waiting for SSH to become available...
==> c1xr06: Waiting for SSH to become available...
==> d2xr02: Waiting for SSH to become available...
==> d1xr01: Waiting for SSH to become available...
Step 3: starting the lab -- libvirt: sleep 10
============================================================
Step 3: starting the lab -- libvirt: vagrant up --provider libvirt d3xr01 d3xr02 d1xfw01 d2xfw01 h1xgw1 h1xgw2 h1xsw01 h1xsw02 h1xsw03 h1xsw04
============================================================
```

But figure if you need to troubleshoot why boxes aren't properly booting, you can turn off batching or boot the individual VM, so I consider this a non-issue.

File: docs/labs/libvirt.md [hunk 1]
```diff
@@ -98,6 +98,31 @@ You can change the parameters of the management network in the **addressing.mgmt
...
+
+* Configure the batch size with **defaults.providers.libvirt.batch_size** parameter (an integer between 1 and 50)
+* Configure idle interval between batches (if needed) with **defaults.providers.libvirt.batch_interval** parameter (between 1 and 1000 seconds).
+
+Example:
+
+```
+provider: libvirt
+defaults.device: cumulus
+defaults.providers.libvirt.batch_size: 2
```
Comment (petercrocker) [inline docs/labs/libvirt.md:113]: I'd suggest a batch size default of a higher number, like 6. Remember people will usually start with your example, and if they use 2 it'll take ages to boot a 40 node topology!

**Review by petercrocker — APPROVED** [review]
Looks good, no additional comments aside from the default batch size example. Code works as promised.

**Comment by ipspace** [comment]
> Note that a very minor issue with this approach is that some of the vagrant status messages end up being suppressed. It seems the next batch kicks in what the all of the batched VMs are booted, and "Waiting for SSH..." but then you don't see the vagrant messages that would normally occur after the VM gets a DHCP IP.

Not my circus, not my monkeys ;) The next command is executed after **vagrant up** completes.

Thanks a million for the review and the testing, will change the default batch size.

**Pull request merged**